Subject: [PATCH] fix build basing on (https://stackoverflow.com/questions/72336864/how-to-compile-openjdk-11-on-macos)
---
Index: src/hotspot/share/jfr/periodic/jfrNetworkUtilization.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hotspot/share/jfr/periodic/jfrNetworkUtilization.cpp b/src/hotspot/share/jfr/periodic/jfrNetworkUtilization.cpp
--- a/src/hotspot/share/jfr/periodic/jfrNetworkUtilization.cpp	(revision a18376494a903ef615c24fa88ff9db7ca11c2182)
+++ b/src/hotspot/share/jfr/periodic/jfrNetworkUtilization.cpp	(revision 338104d1eadd79a4ff776573515171070a0e1577)
@@ -34,6 +34,13 @@
 #include "utilities/globalDefinitions.hpp"
 #include "utilities/growableArray.hpp"
 
+#ifdef assert
+  #undef assert
+  #ifdef vmassert
+    #define assert(p, ...) vmassert(p, __VA_ARGS__)
+  #endif
+#endif
+
 struct InterfaceEntry {
   char* name;
   traceid id;
